core/embed/rust/src/ui/layout_delizia/component/trade_screen.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{
+    strutil::TString,
+    ui::{
+        component::{paginated::PaginateFull, Bar, Component, Event, EventCtx, Label, Never},
+        geometry::Rect,
+        shape::Renderer,
+        util::Pager,
+    },
+};
+
+use super::super::theme;
+
+pub struct TradeScreen {
+    sell_amount: Label<'static>,
+    line: Bar,
+    buy_amount: Label<'static>,
+}
+
+impl TradeScreen {
+    pub fn new(sell_amount: TString<'static>, buy_amount: TString<'static>) -> Self {
+        Self {
+            sell_amount: Label::left_aligned(sell_amount, theme::TEXT_WARNING).top_aligned(),
+            line: Bar::new(theme::GREY_EXTRA_DARK, theme::BG, 1),
+            buy_amount: Label::left_aligned(buy_amount, theme::TEXT_MAIN_GREEN_LIME)
+                .bottom_aligned(),
+        }
+    }
+}
+
+impl PaginateFull for TradeScreen {
```

Comment:
There is a trait `SinglePage` that does the same
```suggestion
impl SinglePage for TradeScreen {}
```

Reply:
Indeed, I don't remember why I switched it back to `PaginateFull` but it's certainly not necessary d4b79bd6a4e6d8b4c8da5604b0d12c664ff9082a